### ISSUE TITLE
Loading inner model optimizer

### ISSUE THREAD

**Issue by maxbry** [keras-team/keras#10835]
I have a pretrained model, that I'm using when training a second model:

    pretrained = load_model('pretrained')
    input = Input(shape)
    var = pretrained(input)
    var = new_model(var)

    model = Model(input, var)
    model.compile([...])

The pretrained model had an optimizer, like sgd with momentum. When loading the model like this and compiling it into a bigger model, will the previous values of the optimizer be used?

**Comment by greed2411** [comment]
That depends on your usage if I'm not wrong, `model.compile()` all it does is build the graph, if you are going to use `model.fit()` yes it's definitely going to retrain the weights with the new optimizer, but if you are going to do is just `model.eval()` or `model.predict()` the graph is going to be set in evaluation mode and optimizer has got nothing to do with it.

Since you are saying you are using it to build a bigger model, whatever the optimizer decides the new weights should be it updates the old values from the old optimizer, if you don't wish for that you could try freezing the layers.
